video_xblock/video_xblock.py

```diff
@@ -0,0 +1,230 @@
+from xblock.fragment import Fragment
+from xblock.plugin import Plugin
+from xblock.validation import ValidationMessage
+
+from django.template import Template, Context
+
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+import HTMLParser
+
+html_parser = HTMLParser.HTMLParser()
+
+_ = lambda text: text
+log = logging.getLogger(__name__)
+
+# From official Wistia documentation. May change in the future
+# https://wistia.com/doc/construct-an-embed-code#the_regex
+VIDEO_URL_RE = re.compile(r'https?:\/\/(.+)?(wistia.com|wi.st)\/(medias|embed)\/.*')
+
+
+class BaseVideoPlayer(Plugin):
+    __metaclass__ = abc.ABCMeta
+
+    entry_point = 'video_xblock.v1'
+
+    @abc.abstractproperty
+    def url_re(self):
+        """
+        Regex (list) to match video url
+        """
+        return [] or re.RegexObject or ''
+
+    @abc.abstractmethod
+    def get_frag(self, **context):
+        return Fragment('<video />')
+
+    @abc.abstractmethod
+    def media_id(self, href):
+        """
+        Extracts Platform's media id from the video url.
+        E.g. https://example.wistia.com/medias/12345abcde -> 12345abcde
+        """
+
+        return ''
+
+    def get_player_html(self, **context):
+        frag = self.get_frag(**context)
+        return Response(
+            frag.head_html()+frag.body_html()+frag.foot_html(),
+            content_type='text/html')
+
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    def match(self, href):
+        if isinstance(self.url_re, list):
+            return any(regex.search(href) for regex in self.url_re)
+        elif isinstance(self.url_re, re.RegexObject):
+            return self.url_re.search(href)
+        elif isinstance(self.url_re, basestring):
+            return re.search(self.url_re, href, re.I)
+
+
+class DummyPlayer(BaseVideoPlayer):
+    @property
+    def url_re(self):
+        return [re.compile(r'')]
+
+    def get_frag(self, **context):
+        return Fragment(u'[Here be Video]')
+
+    def media_id(self, href):
+        return '<media_id>'
+
+
+class YoutubePlayer(BaseVideoPlayer):
+    # http://regexr.com/3a2p0
+    url_re = re.compile(r'(?:youtube\.com\/\S*(?:(?:\/e(?:mbed))?\/|watch\?(?:\S*?&?v\=))|youtu\.be\/)(?P<media_id>[a-zA-Z0-9_-]{6,11})')
+
+    def media_id(self, href):
+        return self.url_re.search(href).group('media_id')
+
+    def get_frag(self, **context):
+        html = Template(self.resource_string("static/html/youtube.html"))
+        frag = Fragment(
+            html_parser.unescape(
+                html.render(Context(context))
+            )
+        )
+
+        frag.add_css(self.resource_string(
+            'static/bower_components/video.js/dist/video-js.min.css'
+        ))
+        frag.add_javascript(self.resource_string(
+            'static/bower_components/video.js/dist/video.min.js'
```

Comment:
I thought `videojs` will be added automatically by VideoXblock to every Backend. What is the reason of duplicating it?